src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/WindowsFormsAddItemFilter.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms
+{
+    /// <summary>
+    /// A filter for the Add New Item dialog that filters out Windows Forms items from non-Windows Forms projects.
+    /// </summary>
+    [ExportProjectNodeComService(typeof(IVsFilterAddProjectItemDlg))]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class WindowsFormsAddItemFilter : IVsFilterAddProjectItemDlg, IDisposable
+    {
+        private  UnconfiguredProject? _project;
+
+        [ImportingConstructor]
+        public WindowsFormsAddItemFilter(UnconfiguredProject project)
+        {
+            _project = project;
+        }
+
+        public int FilterTreeItemByLocalizedName(ref Guid rguidProjectItemTemplates, string pszLocalizedName, out int pfFilter)
+        {
+            pfFilter = 0;
+            return HResult.NotImplemented;
+        }
+
+        public int FilterTreeItemByTemplateDir(ref Guid rguidProjectItemTemplates, string pszTemplateDir, out int pfFilter)
+        {
+            pfFilter = 0;
```

Comment:
Capture the `_project` field an a variable, and if its null, return E_UNEXPECTED so that we don't return OK on null.